### PR TITLE
fix(sbb-dialog): handle unwanted backdrop clicks on overlay components

### LINKED ISSUE #1834: [Bug]: can't use sbb-datepicker and sbb-select in sbb-dialog
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/lyne-design-system/lyne-components/blob/master/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/lyne-design-system/lyne-components/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/orgs/lyne-design-system/projects/4) for an issue that matches the one I want to file, without success.

### Bug type

Functionality

### It affects the following packages

Design Tokens

### Which version of Lyne Components are you using

0.36.13

### Operating system

Win 10

### Browser / Browser version

Chrome 114.0.5735.134

### Input mode

Mouse/Keyboard

### Additional settings

_No response_

### What happened?

When i use the sbb-datepicker in the sbb-dialog modal context and choose a day from the datepicker inside an sbb-form-field the whole dialog is closing.

### Steps to reproduce the issue

1. open dialog
2. click on the datepicker input
3. select a day

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by MarioCastigliano**
## Preflight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/lyne-design-system/lyne-components/blob/master/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/lyne-design-system/lyne-components/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [pull request tracker](https://github.com/lyne-design-system/lyne-components/pulls) for a Pull Request (PR) that matches the one I want to submit, without success.

## Issue

This PR Closes #1834 

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)

See [Review Guidelines](../REVIEW.md) for more information what is checked during review process.

## Changes

Changes in this pull request:

- Check event's composedPath to make sure click on overlays that overflow the dialog won't be counted as clicks on the backdrop

## Browsers

I tested the build on the following browsers:

- [ ] Firefox Desktop
- [x] Chrome Desktop
- [ ] Edge Desktop
- [ ] Safari Desktop
- [ ] Chrome Mobile
- [ ] Safari Mobile

## Screen readers

I tested the build on the following browsers:

- [ ] JAWS Firefox Desktop
- [ ] JAWS Chrome Desktop
- [ ] NVDA Firefox Desktop
- [ ] NVDA Chrome Desktop
- [ ] VoiceOver Safari Desktop
- [ ] VoiceOver Chrome Desktop
- [ ] VoiceOver Safari Mobile
- [ ] Android Accessibility Suite Chrome Mobile

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/components/sbb-dialog/sbb-dialog.tsx [hunk 1]
```diff
@@ -254,12 +253,21 @@ export class SbbDialog implements ComponentInterface {
 
   // Check if the pointerdown event target is triggered on the dialog.
   private _pointerDownListener = (event: PointerEvent): void => {
-    this._isPointerDownEventOnDialog = isEventOnElement(this._dialog, event);
+    this._isPointerDownEventOnDialog = event
+      .composedPath()
+      .filter((e) => e instanceof window.HTMLElement)
+      .some((target) => (target as HTMLElement).tagName === 'DIALOG');
```
Comment (jeripeierSBB) [inline src/components/sbb-dialog/sbb-dialog.tsx:259]: Not sure if we maybe should give the dialog an id to be more refactoring stable? But however, some automatic tests would help to keep this improvement stable, can you create 1 to 2 e2e tests?

**Review by jeripeierSBB — CHANGES_REQUESTED** [review]
Thanks for addressing this!

File: src/components/sbb-dialog/sbb-dialog.tsx [hunk 2]
```diff
@@ -254,12 +256,21 @@ export class SbbDialog implements ComponentInterface {
 
   // Check if the pointerdown event target is triggered on the dialog.
   private _pointerDownListener = (event: PointerEvent): void => {
-    this._isPointerDownEventOnDialog = isEventOnElement(this._dialog, event);
+    this._isPointerDownEventOnDialog = event
+      .composedPath()
+      .filter((e) => e instanceof window.HTMLElement)
```
Comment (kyubisation) [inline src/components/sbb-dialog/sbb-dialog.tsx:261]: Nitpick: You could use typescript assertion to not have to cast to HTMLElement in some
```suggestion
      .filter((e): e is HTMLElement => e instanceof window.HTMLElement)
```

File: src/components/sbb-dialog/sbb-dialog.tsx [hunk 3]
```diff
@@ -254,12 +256,21 @@ export class SbbDialog implements ComponentInterface {
...
   };
 
   // Close dialog on backdrop click.
   private _closeOnBackdropClick = async (event: PointerEvent): Promise<void> => {
-    if (!this._isPointerDownEventOnDialog && !isEventOnElement(this._dialog, event)) {
+    if (
+      !this._isPointerDownEventOnDialog &&
+      !event
+        .composedPath()
+        .filter((e) => e instanceof window.HTMLElement)
```
Comment (kyubisation) [inline src/components/sbb-dialog/sbb-dialog.tsx:271]: Nitpick: You could use typescript assertion to not have to cast to HTMLElement in some
```suggestion
        .filter((e): e is HTMLElement => e instanceof window.HTMLElement)
```

**Review by kyubisation — APPROVED** [review]
LGTM 👍 
Thanks for the fix 😃

**Pull request merged**